### DISCUSSION TITLE
Context Provider Compose

### DISCUSSION THREAD

**Discussion by EricRovell** [solidjs/solid#2246] [Q&A] [answered]
I use `Context API` quite a lot within my application and decided to write an utility to compose providers where there are too many of them.

My solution looks like this:

```tsx
import type { ParentComponent, ParentProps} from "solid-js";

const ProviderCompose = (props: Props) => (
	<>
		{props.components.reduceRight((acc, Component) => <Component>{acc}</Component>, props.children)}
	</>
);
```

And the usage like this:

```tsx
export const App = () => {

	return (
		<>
			<ProviderCompose components={[ MetaProvider, SettingsProvider, TranslationProvider ]}>
				<Routes />
			</ProviderCompose>
		</>
	);
}
```

But the problem is, it does not work. The error appears within `useContext` hook: `Error: useSettings must be used with a Settings.Provider`. This seems strange, am I doing something wrong by passing the children? Thanks for help!

**Comment by atk** [comment]
There's the `@solid-primitives/context` package that makes working with context easier.

**Reply by EricRovell** [reply, marked as answer]
> There's the `@solid-primitives/context` package that makes working with context easier.

I see, there is [MultiProvider](https://primitives.solidjs.community/package/context#multiprovider) utility. I will look how it is implemented, thanks!
